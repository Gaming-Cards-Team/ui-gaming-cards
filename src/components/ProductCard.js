import React, { Suspense } from "react";
import Card from 'react-bootstrap/Card';
import Loader from 'react-loader-spinner'

const cardStyle = {
  paddingTop: 10,
  paddingBottom: 15,
};

const ImageCard = React.lazy(() => import('./ImageCard'))

const ProductCard = (props) => {
  const { image } = props;
  return (
    <div style={cardStyle}>
        <Card >
          <Card.Body>
              <Suspense fallback={<Loader/>}>
                <ImageCard image={image} />
              </Suspense>
          </Card.Body> 
        </Card>
    </div>
  );
};

export default ProductCard;
