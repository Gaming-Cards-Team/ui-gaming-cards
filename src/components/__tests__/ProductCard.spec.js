import React from "react";
import ProductCard from "../ProductCard";
import renderer from "react-test-renderer";

describe("Product Card component", () => {
  it("should render successfully", () => {
    const component = renderer.create(<ProductCard />).toJSON();

    expect(component).toMatchSnapshot();
  });
});