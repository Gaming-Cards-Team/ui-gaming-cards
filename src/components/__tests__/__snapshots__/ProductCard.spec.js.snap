// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product Card component should render successfully 1`] = `
<div
  style={
    Object {
      "paddingBottom": 15,
      "paddingTop": 10,
    }
  }
>
  <div
    className="card"
  >
    <div
      className="card-body"
    >
      <span
        className=" lazy-load-image-background blur"
        style={
          Object {
            "backgroundImage": "",
            "backgroundSize": "",
            "color": "transparent",
            "display": "inline-block",
            "height": undefined,
            "width": "100%",
          }
        }
      >
        <span
          className=""
          style={
            Object {
              "display": "inline-block",
              "width": "100%",
            }
          }
        />
      </span>
    </div>
  </div>
</div>
`;
